resources/migrations/001_update_migrations.yaml

```diff
@@ -6859,6 +6859,67 @@ databaseChangeLog:
             columnDataType: integer
       rollback: # we can't reliably add back the constraint while downgrading
 
+  - changeSet:
+      id: v50.2024-05-08T09:00:00
+      author: qnkhuat
+      comment: Add task_history.status
+      changes:
+        - addColumn:
+            tableName: task_history
+            columns:
+              - column:
+                  name: status
+                  type: varchar(21)
+                  remarks: "the status of task history, could be started, failed, success, unknown"
+                  constraints:
+                    nullable: false
+                  defaultValue: "unknown"
```

Comment:
Not a blocker, but from your description it sounds like we could infer "success" for many of the existing rows.

Reply:
It would also be nice if we dropped this default in a later migration to prevent accidentally forgetting to initialize it for new records.

Reply:
> Not a blocker, but from your description, we could infer "success" for many of the existing rows.

Yea, I did ponder on that. To do that, we'll need to extract `details` to see if there are any `{:status "failed"}.` Considering it's a big table and the old TaskHistory will be deleted anyway, I think the effort outweighs the benefit here.

Reply:
> It would also be nice if we dropped this default in a later migration to prevent accidentally forgetting to initialize it for new records.

good idea, I made `started` the default value in 
891e6cd6a7